File: src/main/java/railgunDownloaderV4/Application.java
/*
 * RAILGUN DOWNLOADER - VERSION 4.0.0
 * Application.java
 */
package railgunDownloaderV4;
import railgunDownloaderV4.components.Facebook;
import railgunDownloaderV4.components.MangaDex;
import railgunDownloaderV4.components.Youtube;
import railgunDownloaderV4.components.Nhentai;

import javax.swing.*;
import java.awt.*;
import java.util.Objects;

public class Application {
    public JFrame App;
    private final Youtube YoutubeComponent;
    private final Nhentai nhentaiComponent;
    private final Facebook facebookComponent;
    private final MangaDex mangaDexComponent;

    private final JButton YoutubeButton;
    private final JButton nhentaiButton;
    private final JButton facebookButton;
    private final JButton mangaDexButton;


    public Application() {
        App = new JFrame();
        YoutubeComponent = new Youtube(this);
        nhentaiComponent = new Nhentai(this);
        facebookComponent = new Facebook(this);
        mangaDexComponent = new MangaDex(this);

        YoutubeButton = new JButton();
        nhentaiButton = new JButton();
        facebookButton = new JButton();
        mangaDexButton = new JButton();
    }
    public static void main(String[] args) {
        SwingUtilities.invokeLater(() -> {
            Application AppScene = new Application();
            AppScene.SetProperty();
            AppScene.YoutubeComponent.SetYoutubeButton(AppScene.YoutubeButton);
            AppScene.nhentaiComponent.setNhentaiUIButton(AppScene.nhentaiButton);
            AppScene.facebookComponent.setFacebookButton(AppScene.facebookButton);
            AppScene.mangaDexComponent.setMangaDex(AppScene.mangaDexButton);
        });
    }

    private void SetProperty() {
        App.setTitle("Railgun Downloader V4");
        App.setIconImage(
               Objects.requireNonNull(
                       Toolkit.getDefaultToolkit()
                               .getImage(getClass().getResource("/AppIcon.jpg"))
               )
        );
        App.setSize(600, 600);
        App.setMaximumSize(new Dimension(600, 600));
        App.setLayout(null);
        App.setLocationRelativeTo(null);
        App.getContentPane().setBackground(new Color(21, 21, 21));
        App.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
        App.setResizable(false);
        App.setVisible(true);
    }
}

File: src/main/kotlin/railgunDownloaderV4/Application.kt
/*
 * RAILGUN DOWNLOADER - VERSION 4.0.0
 * File: Application.kt
 * This project is licensed under GPL-3.0 License.
 * LICENSE: https://www.gnu.org/licenses/gpl-3.0.html
 * Contribution: Reim
 */
package railgunDownloaderV4

import railgunDownloaderV4.components.Facebook
import railgunDownloaderV4.components.MangaDex
import railgunDownloaderV4.components.Nhentai
import railgunDownloaderV4.components.Youtube
import railgunDownloaderV4.components.ulti.SetIcon
import java.awt.Color
import java.awt.Dimension
import javax.swing.JButton
import javax.swing.JFrame
import javax.swing.SwingUtilities
import javax.swing.WindowConstants

class Application {
    val app: JFrame by lazy { JFrame() }
    private val youtube: Youtube by lazy { Youtube(this) }
    private val nhentaiComponent: Nhentai by lazy { Nhentai(this) }
    private val facebookComponent: Facebook by lazy { Facebook(this) }
    private val mangaDexComponent: MangaDex by lazy { MangaDex(this) }
    private val setIcon: SetIcon by lazy { SetIcon() }

    private val youtubeButton: JButton by lazy { JButton() }
    private val nhentaiButton: JButton by lazy { JButton() }
    private val facebookButton: JButton by lazy { JButton() }
    private val mangaDexButton: JButton by lazy { JButton() }

    private fun setProperty() {
        app.title = "Railgun Downloader V4"
        app.iconImage = setIcon.setAppIcon()
        app.setSize(600, 600)
        app.maximumSize = Dimension(600, 600)
        app.layout = null
        app.setLocationRelativeTo(null)
        app.contentPane.background = Color(21, 21, 21)
        app.defaultCloseOperation = WindowConstants.EXIT_ON_CLOSE
        app.isResizable = false
        app.isVisible = true
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            SwingUtilities.invokeLater {
                val appScene = Application()
                appScene.setProperty()
                appScene.youtube.setYoutubeButton(appScene.youtubeButton)
                appScene.nhentaiComponent.setNhentaiUIButton(appScene.nhentaiButton)
                appScene.facebookComponent.setFacebookButton(appScene.facebookButton)
                appScene.mangaDexComponent.setMangaDex(appScene.mangaDexButton)
            }
        }
    }
}
